File: benchmark.py
import subprocess
import time

# Python
python_start_time = time.time()
subprocess.run(["python", "python_script.py"])
python_elapsed_time = time.time() - python_start_time

# JavaScript (Node.js)
javascript_start_time = time.time()
subprocess.run(["node", "javascript_script.js"])
javascript_elapsed_time = time.time() - javascript_start_time

# Rust
rust_start_time = time.time()
subprocess.run(["./target/release/rust_executable"])
rust_elapsed_time = time.time() - rust_start_time

# C++
cpp_start_time = time.time()
subprocess.run(["./cpp_executable"])
cpp_elapsed_time = time.time() - cpp_start_time

print("Python Elapsed Time:", python_elapsed_time, "seconds")
print("JavaScript Elapsed Time:", javascript_elapsed_time, "seconds")
print("Rust Elapsed Time:", rust_elapsed_time, "seconds")
print("C++ Elapsed Time:", cpp_elapsed_time, "seconds")

File: benchmark.js
const { execSync } = require('child_process');

// Python
const pythonStartTime = Date.now();
execSync('python python_script.py', { stdio: 'inherit' });

const pythonElapsedTime = Date.now() - pythonStartTime;
console.log(`Python Elapsed Time: ${pythonElapsedTime / 1000} seconds`);

// JavaScript (Node.js)

const javascriptStartTime = Date.now();
execSync('node javascript_script.js', { stdio: 'inherit' });

const javascriptElapsedTime = Date.now() - javascriptStartTime;
console.log(`JavaScript Elapsed Time: ${javascriptElapsedTime / 1000} seconds`);


// Rust
const rustStartTime = Date.now();
execSync('./target/release/rust_executable', { stdio: 'inherit' });

const rustElapsedTime = Date.now() - rustStartTime;
console.log(`Rust Elapsed Time: ${rustElapsedTime / 1000} seconds`);

// C++
const cppStartTime = Date.now();
execSync('./cpp_executable', { stdio: 'inherit' });

const cppElapsedTime = Date.now() - cppStartTime;
console.log(`C++ Elapsed Time: ${cppElapsedTime/ 1000} seconds`);
